test/cases/api.sh

```diff
@@ -1039,6 +1206,109 @@ function verifyDisk() {
     greenprint "✅ ${filename} image info verified"
 }
 
+# Verify VMDK image in VSphere
+function verifyInVSphere() {
+    local _filename="$1"
+    greenprint "Verifying VMDK image: ${_filename}"
+
+    # Create SSH keys to use
+    local _vsphere_ssh_key="${WORKDIR}/vsphere_ssh_key"
+    ssh-keygen -t rsa-sha2-512 -f "${_vsphere_ssh_key}" -C "${SSH_USER}" -N ""
+
+    VSPHERE_VM_NAME="osbuild-composer-vm-${TEST_ID}"
+
+    # create cloud-init ISO with the configuration
+    local _ci_userdata_path
+    _ci_userdata_path="$(_createCIUserdata "${SSH_USER}" "${_vsphere_ssh_key}.pub")"
+    local _ci_metadata_path
+    _ci_metadata_path="$(_createCIMetadata "${VSPHERE_VM_NAME}")"
+    greenprint "💿 Creating cloud-init user-data ISO"
+    local _ci_iso_path
+    _ci_iso_path="$(_createCIUserdataISO "${_ci_userdata_path}" "${_ci_metadata_path}")"
+
+    VSPHERE_IMAGE_NAME="${VSPHERE_VM_NAME}.vmdk"
+
+    # import the built VMDK image to VSphere
+    # import.vmdk seems to be creating the provided directory and
+    # if one with this name exists, it appends "_<number>" to the name
+    greenprint "🚧 Converting the downloaded VMDK image to be streamOptimized"
+    qemu-img convert -O vmdk -o subformat=streamOptimized "${_filename}" "${WORKDIR}/${VSPHERE_IMAGE_NAME}"
```

Comment:
With your last comment, is this still needed? Or did you not rebase yet?

Reply:
I added one new commit which implemented this change https://github.com/osbuild/osbuild-composer/pull/2459/commits/5c46e0c9fe7e49407eb07b14c7f7c5a3e7193586. Mainly because it seemed cleaner to me.

Reply:
Ok! I'll re-review the PR as a whole once not in draft :)